pr/land.go

```diff
@@ -2,14 +2,103 @@ package pr
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"sync"
 
+	"github.com/abhinav/git-fu/gateway"
 	"github.com/abhinav/git-fu/service"
+
+	"github.com/google/go-github/github"
+	"go.uber.org/multierr"
 )
 
+func (s *Service) checkApproved(ctx context.Context, pr *github.PullRequest) error {
+	reviews, err := s.GitHub.ListPullRequestReviews(ctx, pr.GetNumber())
+	if err != nil {
+		return err
+	}
+
+	for _, review := range reviews {
+		if review.Status != gateway.PullRequestChangesRequested {
+			continue
+		}
+
+		err = multierr.Append(err,
```

Comment:
I think you want the last state for a user, since the same user can have multiple reviews, 1 where they asked for changes and then a later one where they accept. It looks like the API does return all of them, so you'll want to maintain a map of user id -> lastState